src/transformers/processing_utils.py

```diff
@@ -830,6 +827,60 @@ class MyProcessingKwargs(ProcessingKwargs, CommonKwargs, TextKwargs, ImagesKwarg
             output_kwargs[modality].update(output_kwargs["common_kwargs"])
         return output_kwargs
 
+    def _check_reversed_images_text(self, images, text):
+        """
+        For backward compatibility: reverse the order of `images` and `text` inputs if they are swapped.
+        This method should only be called for processors where `images` and `text` have been swapped for uniformization purposes.
+        Note that this method assumes that two `None` inputs are valid inputs. If this is not the case, it should be handled
+        in the processor's `__call__` method before calling this method.
+        """
+
+        def _is_valid_text_input(t):
+            if isinstance(t, str):
+                # Strings are fine
+                return True
+            elif isinstance(t, (list, tuple)):
+                # List are fine as long as they are...
+                if len(t) == 0:
+                    # ... empty
+                    return True
+                elif isinstance(t[0], str):
+                    # ... list of strings
+                    return True
+                elif isinstance(t[0], (list, tuple)):
+                    # ... list with an empty list or with a list of strings
+                    return len(t[0]) == 0 or isinstance(t[0][0], str)
+                else:
+                    return False
+            else:
+                return False
+
+        def _is_valid_or_convertible(input, validator, converter):
+            is_valid = validator(input) or input is None
+            is_convertible = converter(input) if not is_valid else False
+            return is_valid, is_convertible
+
+        images_is_valid, images_is_text = _is_valid_or_convertible(images, valid_images, _is_valid_text_input)
+        text_is_valid, text_is_images = _is_valid_or_convertible(text, _is_valid_text_input, valid_images)
+
+        # Handle cases where both inputs are valid
+        if images_is_valid and text_is_valid:
+            return images, text
+
+        # Handle cases where inputs need to and can be swapped
+        if (
+            (images is None and text_is_images)
+            or (text is None and images_is_text)
+            or (images_is_text and text_is_images)
+        ):
+            logger.warning_once(
+                "You may have used the wrong order for inputs. `images` should be passed before `text`. "
+                "The `images` and `text` inputs will be swapped."
```

Comment:
Should we include info about deprecating this behavior in a future version here?